### PR TITLE
fix make_nr if no dups exist

### PR DESCRIPTION
**Pull request by bernt-matthias**
the branch for the case that no duplicates exist
fails because is assumes that the input is a
single file:

```
Traceback (most recent call last):
  File "/gpfs1/data/galaxy_server/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/peterjc/make_nr/c84f12187af9/make_nr/tools/make_nr/make_nr.py", line 133, in <module>
    make_nr(args, options.output, options.sep, options.alphasort)
  File "/gpfs1/data/galaxy_server/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/peterjc/make_nr/c84f12187af9/make_nr/tools/make_nr/make_nr.py", line 128, in make_nr
    os.symlink(os.path.abspath(input_fasta), output_fasta)
  File "/gpfs1/data/galaxy_server/galaxy/database/dependencies/_conda/envs/__biopython@1.67/lib/python3.6/posixpath.py", line 378, in abspath
    path = os.fspath(path)
TypeError: expected str, bytes or os.PathLike object, not list
```

### CODE REVIEW

**Comment by bernt-matthias** [comment]
ping @peterjc .. not sure if I want to fix the unrelated linting errors

**Comment by peterjc** [comment]
Looks like a flake8 plugin update got stricter (B902):

```
$ flake8 .
./data_managers/ncbi_blastdb/fetch_blast_db.py:103:1: B902 blind except Exception: statement
./datatypes/blast_datatypes/blast.py:183:1: B902 blind except Exception: statement
./datatypes/blast_datatypes/blast.py:221:1: B902 blind except Exception: statement
./tools/make_nr/make_nr.py:14:1: F401 'os' imported but unused
./tools/ncbi_blast_plus/blastxml_to_tabular.py:194:1: B902 blind except Exception: statement
./tools/ncbi_blast_plus/blastxml_to_tabular.py:201:1: B902 blind except Exception: statement
./tools/blastxml_to_top_descr/blastxml_to_top_descr.py:173:1: B902 blind except Exception: statement
./tools/blastxml_to_top_descr/blastxml_to_top_descr.py:184:1: B902 blind except Exception: statement
./tools/blast2go/blast2go.py:81:1: B902 blind except Exception: statement
```

I should fix those separately... however this one is real and needs fixing:

```
./tools/make_nr/make_nr.py:14:1: F401 'os' imported but unused
```

**Comment by peterjc** [comment]
See #138 for the flake8 blind exceptions, should get merged soon...

**Comment by peterjc** [comment]
Note to self: Add a test for this (especially as there are other issues needing work in this script)

**Comment by bernt-matthias** [comment]
> Note to self: Add a test for this (especially as there are other issues needing work in this script)

lets see if it is as easy as I hope it is :)

File: tools/make_nr/make_nr.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def make_nr(input_fasta, output_fasta, sep=";", sort_ids=False):
         )
     elif len(input_fasta) == 1:
         # Single file, no need to make a copy or edit titles
-        os.symlink(os.path.abspath(input_fasta), output_fasta)
```
Comment (peterjc) [inline tools/make_nr/make_nr.py:143]: Doh. I clearly wasn't thinking there. I was right to want to add a test though 🤣

**Comment by peterjc** [comment]
Well, that's interesting:

```
 Traceback (most recent call last):
|  File "/home/travis/.planemo/planemo_tmp_nbrx7k2t/make_nr/make_nr.py", line 158, in <module>
|  make_nr(args, options.output, options.sep, options.alphasort)
|  File "/home/travis/.planemo/planemo_tmp_nbrx7k2t/make_nr/make_nr.py", line 144, in make_nr
|  os.symlink(os.path.abspath(input_fasta[0]), output_fasta)
|  FileExistsError: [Errno 17] File exists: '/tmp/tmp_b8190y4/files/4/f/6/dataset_4f6c366a-9e18-44d4-9d92-3a28fadcd3b0.dat' -> '/tmp/tmp_b8190y4/files/0/9/c/dataset_09c92870-8f84-4d13-b25f-c0b39774ae8b.dat'
```

My guess without trying this and a vague Galaxy memory, is that Galaxy creates an empty file with the expect output name prior to calling the tool. Thus making a symlink fails. Surely not?

(If so, there was more than one bug in this untested branch of the 'simple' script)

**Comment by bernt-matthias** [comment]
Jep. We need to copy the file.

File: tools/make_nr/make_nr.xml [hunk 2]
```diff
@@ -51,6 +51,14 @@ python $__tool_directory__/make_nr.py $alphasort -s '$separator' -o '$output'
...
             <output name="output" file="deduplicate.sortids.fasta" ftype="fasta"/>
         </test>
+        <test>
+            <param name="input" value="empty.fasta" ftype="fasta"/>
+            <output name="output" file="empty.fasta" ftype="fasta"/>
+        </test>
+        <test>
+            <param name="input" value="empty.fasta,empty.fasta" ftype="fasta"/>
+            <output name="output" file="empty.fasta" ftype="fasta"/>
+        </test>
```
Comment (peterjc) [inline tools/make_nr/make_nr.xml:61]: That's elegant ❤️

But we'll need to add ``empty.fasta`` to the tool manifest in ``tools/make_nr/.shed.yml``

Comment (bernt-matthias) [reply tools/make_nr/make_nr.xml:61]: will do. lets see if this elegant solution is also a working solution :)

**Comment by peterjc** [comment]
That's live now on http://testtoolshed.g2.bx.psu.edu/view/peterjc/make_nr and http://toolshed.g2.bx.psu.edu/view/peterjc/make_nr - thank you!

**Pull request merged**